ignite/contrib/handlers/param_scheduler.py

```diff
@@ -980,10 +1005,7 @@ def __init__(self, schedulers: List[ParamScheduler], names: Optional[List[str]]
 
         self.schedulers = schedulers
         self.names = names
-
-        self.optimizer = self.schedulers[0].optimizer
-        if not (all(id(s.optimizer) == id(self.optimizer) for s in schedulers)):
-            raise ValueError("schedulers should be related to same optimizer")
+        self.optimizer = _ParamGroupsOptimizer(schedulers)
```

Comment:
I think I missed that in your previous PR. But why `ParamGroupScheduler` is derived from `ParamScheduler` now ?
I think it does not match correctly the fields like `self.optimizer` or `self.optimizer_param_groups` etc as it can have multiple optimizers and groups at the same time...

Reply:
Yes, it is derived from. 
`self.optimizer` is now an object `_ParamGroupsOptimizer` encapsulating multiple optimizers. It works because API allows objects with `param_groups`. In fine, `ParamGroupScheduler` packs optimizers and relevant `param_groups` from them. I think that is correct.

Reply:
I thought about the implementation. I feel you are right, the problem is related to `ParamGroupsScheduler` derived from `ParamScheduler`. It's easy to write buggy code.
The point that I have to solve is the implementation of `create_lr_scheduler_with_warmup` which mix `ParamGroupsScheduler` and `ConcatScheduler`. I have an idea about that. Let's see.

Reply:
Thanks @sdesrozis !

Reply:
Ok I think i got it. 

To have multi params group in warmup (issue 856), I modified `ParamGroupsScheduler` to work with `ConcatScheduler`. I think the idea was good but not the actual implementation... First, `ParamGroupsScheduler` (used with `ConcatScheduler`) should be related to the same optimizer. In that case, only one call of `scheduler()` should be done (maybe a bug here). Second, it should inherit from `ParamScheduler` and pass the optimizer.

In other hand, we need an implementation of `ParamGroupsScheduler` that packs schedulers related to different optimizers. In fact, we must be sure that schedulers with same optimizers are not embbeded to avoid mutliple calls in one call... (I will check that).

@vfdev-5 thoughts ? Sorry, this item is a bit tricky.